libscript/src/sort.mlc

```diff
@@ -245,4 +245,66 @@ end syntax
 
 --
 
+/*
+Summary: A handler type that can be used in the <SortListUsingHandler> command.
+
+Parameters:
+pLeft: The left hand value to compare.
+pRight: The right hand value to compare.
+
+Returns: 
+An integer less than 0 if pLeft is less than pRight with respect to the intended ordering,
+an integer greater than 0 if pLeft is greater than pRight with respect to the intended ordering,
+or 0 if pLeft is equal to pRight.
+
+Description:
+Any handler of type <SortCompare> can be passed to <SortListUsingHandler> to sort a list. 
+It takes two arguments, and returns an integer based on the comparison.
+
+Tags: Sorting
+*/
+public handler type SortCompare(in pLeft as any, in pRight as any) returns Integer
+
+public foreign handler MCSortExecSortListUsingHandler(inout Target as List, in Handler as SortCompare) returns nothing binds to "<builtin>"
+
+/*
+Summary:        Sorts <Target> using <Handler> as a comparison function.
+Target:         An expression that evaluates to a list.
+Handler: 		A handler of type <SortCompare>
+
+Example:
+ handler RandomSort(in pLeft as any, in pRight as any) returns Integer
+ 	if any number > 0.5 then
+ 		return 1
+ 	else
+ 		return -1
+ 	end if
+ end handler
```

Comment:
Depends what you mean by 'valid' - it will do something :smile: 

But yes, I take your point, something more useful might be more ... useful.